### ISSUE TITLE
Clarification on Dataset File Structure & Training Template Request

### ISSUE THREAD

**Issue by daxuezha** [huaqlili/unsupervised-light-enhance-ICLR2025#4]
Thank you for sharing your code. I have a couple of questions regarding its structure and usage:

Dataset File Structure: Could you please clarify how the dataset is organized? I'm particularly interested in understanding the recommended directory structure and any specific naming conventions for the files.

Training Template: Is there a training template or an example script available that demonstrates how to set up and run training with this dataset? Having a template would be very helpful in getting started quickly.

Thanks for your time and assistance!

Best regards,
A Beginner.

**Comment by tiankongkongkongruye** [comment]
Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?

**Comment by daxuezha** [comment]
> Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?

I feel that unsupervised learning is like this: I have run many classic unsupervised or self-supervised learning algorithms, but very few have achieved the results shown in the papers, and some even performed worse.

**Comment by tiankongkongkongruye** [comment]
Oh, I see. Thank you for your reply.

**Comment by huaqlili** [comment]
> Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?

Thank you for your attention. Since the version submitted before was the ablation experiment, the function used for downsampling was incorrect, so I have modified the code. The author also recommends using different random seeds and adjusting the ratio of the loss function to achieve better results.

**Comment by Small-flypig** [comment]
> > Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?
> 
> I feel that unsupervised learning is like this: I have run many classic unsupervised or self-supervised learning algorithms, but very few have achieved the results shown in the papers, and some even performed worse.

When reproducing the paper, it was found that the PSNR and SSIM of the paper first increased and then decreased, rather than converging towards the direction of increasing PSNR. Is this a common situation in self-supervised methods?

**Comment by wu1112227676** [comment]
> > > Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?
> > 
> > 
> > I feel that unsupervised learning is like this: I have run many classic unsupervised or self-supervised learning algorithms, but very few have achieved the results shown in the papers, and some even performed worse.
> 
> When reproducing the paper, it was found that the PSNR and SSIM of the paper first increased and then decreased, rather than converging towards the direction of increasing PSNR. Is this a common situation in self-supervised methods?


Hello, I want to ask what PSNR value your training can reach. Why does the PSNR keep decreasing during my training, and it only ends up around 10 in the end?

**Comment by wu1112227676** [comment]
> Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?

Hello, I want to ask what PSNR value your training can reach. Why does the PSNR keep decreasing during my training, and it only ends up around 10 in the end?

**Comment by Small-flypig** [comment]
> > > > Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?
> > > 
> > > 
> > > I feel that unsupervised learning is like this: I have run many classic unsupervised or self-supervised learning algorithms, but very few have achieved the results shown in the papers, and some even performed worse.
> > 
> > 
> > When reproducing the paper, it was found that the PSNR and SSIM of the paper first increased and then decreased, rather than converging towards the direction of increasing PSNR. Is this a common situation in self-supervised methods?
> 
> Hello, I want to ask what PSNR value your training can reach. Why does the PSNR keep decreasing during my training, and it only ends up around 10 in the end?

We selected the checkpoint/epoch with the highest PSNR, as done in the paper. In my experiments, I reached performance close to the paper’s results around epoch 29, but the PSNR then gradually decreased. By the final epoch, it was about 17.

**Comment by wu1112227676** [comment]
> > > > > Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?
> > > > 
> > > > 
> > > > I feel that unsupervised learning is like this: I have run many classic unsupervised or self-supervised learning algorithms, but very few have achieved the results shown in the papers, and some even performed worse.
> > > 
> > > 
> > > When reproducing the paper, it was found that the PSNR and SSIM of the paper first increased and then decreased, rather than converging towards the direction of increasing PSNR. Is this a common situation in self-supervised methods?
> > 
> > 
> > Hello, I want to ask what PSNR value your training can reach. Why does the PSNR keep decreasing during my training, and it only ends up around 10 in the end?
> 
> We selected the checkpoint/epoch with the highest PSNR, as done in the paper. In my experiments, I reached performance close to the paper’s results around epoch 29, but the PSNR then gradually decreased. By the final epoch, it was about 17.

May I ask if you modified the author's code? Why is it that when I train with LOLv1 in the same way, the PSNR starts at 13 and then just stays at 10? In my command line for training, I used `python main.py --save_folder weights_3 --logroot logs/your_log/`, while the author's original command is `python main.py --save_folder weights/LOLv1/ --logroot logs/your_log/ --data_train your_train_data --data_val your_val_data --referance_val the_high_quality_val_data --lr 1e-5 --light_patch 64 --loss_weights [1, 0.1, 0.1, 0.5]`. I removed all the other parameters from it—could that be the reason?

**Comment by Small-flypig** [comment]
> > > > > > Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?
> > > > > 
> > > > > 
> > > > > I feel that unsupervised learning is like this: I have run many classic unsupervised or self-supervised learning algorithms, but very few have achieved the results shown in the papers, and some even performed worse.
> > > > 
> > > > 
> > > > When reproducing the paper, it was found that the PSNR and SSIM of the paper first increased and then decreased, rather than converging towards the direction of increasing PSNR. Is this a common situation in self-supervised methods?
> > > 
> > > 
> > > Hello, I want to ask what PSNR value your training can reach. Why does the PSNR keep decreasing during my training, and it only ends up around 10 in the end?
> > 
> > 
> > We selected the checkpoint/epoch with the highest PSNR, as done in the paper. In my experiments, I reached performance close to the paper’s results around epoch 29, but the PSNR then gradually decreased. By the final epoch, it was about 17.
> 
> May I ask if you modified the author's code? Why is it that when I train with LOLv1 in the same way, the PSNR starts at 13 and then just stays at 10? In my command line for training, I used `python main.py --save_folder weights_3 --logroot logs/your_log/`, while the author's original command is `python main.py --save_folder weights/LOLv1/ --logroot logs/your_log/ --data_train your_train_data --data_val your_val_data --referance_val the_high_quality_val_data --lr 1e-5 --light_patch 64 --loss_weights [1, 0.1, 0.1, 0.5]`. I removed all the other parameters from it—could that be the reason?

英文翻译如下：

> There might be some issues with your reproduction. When I reproduced the original code, the best result I could get was around **19 dB**, but it still didn’t reach the results reported in the paper. You need to modify the parameters of the **exposure loss**; follow the command provided by the authors. I’d recommend taking a look at  https://github.com/huaqlili/unsupervised-light-enhance-ICLR2025/issues/11.

**Comment by wu1112227676** [comment]
> > > > > > > Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?
> > > > > > 
> > > > > > 
> > > > > > I feel that unsupervised learning is like this: I have run many classic unsupervised or self-supervised learning algorithms, but very few have achieved the results shown in the papers, and some even performed worse.
> > > > > 
> > > > > 
> > > > > When reproducing the paper, it was found that the PSNR and SSIM of the paper first increased and then decreased, rather than converging towards the direction of increasing PSNR. Is this a common situation in self-supervised methods?
> > > > 
> > > > 
> > > > Hello, I want to ask what PSNR value your training can reach. Why does the PSNR keep decreasing during my training, and it only ends up around 10 in the end?
> > > 
> > > 
> > > We selected the checkpoint/epoch with the highest PSNR, as done in the paper. In my experiments, I reached performance close to the paper’s results around epoch 29, but the PSNR then gradually decreased. By the final epoch, it was about 17.
> > 
> > 
> > May I ask if you modified the author's code? Why is it that when I train with LOLv1 in the same way, the PSNR starts at 13 and then just stays at 10? In my command line for training, I used , while the author's original command is . I removed all the other parameters from it—could that be the reason?`python main.py --save_folder weights_3 --logroot logs/your_log/``python main.py --save_folder weights/LOLv1/ --logroot logs/your_log/ --data_train your_train_data --data_val your_val_data --referance_val the_high_quality_val_data --lr 1e-5 --light_patch 64 --loss_weights [1, 0.1, 0.1, 0.5]`
> 
> 英文翻译如下：
> 
> > There might be some issues with your reproduction. When I reproduced the original code, the best result I could get was around **19 dB**, but it still didn’t reach the results reported in the paper. You need to modify the parameters of the **exposure loss**; follow the command provided by the authors. I’d recommend taking a look at  [#11](https://github.com/huaqlili/unsupervised-light-enhance-ICLR2025/issues/11).

But if I follow the author's instructions, options like --light_patch 64 and --loss_weights [1, 0.1, 0.1, 0.5] are not defined in the code, so running them would cause errors. Therefore, I multiplied the --loss_weights [1, 0.1, 0.1, 0.5] command-line parameter with the four losses in order, and I simply removed light_patch. 

Also, may I ask what it means to change the exposure loss?

**Comment by Small-flypig** [comment]
> > > > > > > > Hello, I'm training the LOLV1 dataset on an RTX 4090 using the code provided by the author. No matter how I train it, I can't achieve the PSNR, SSIM, and LPIPS data given in the author's paper for testing.What's going on?
> > > > > > > 
> > > > > > > 
> > > > > > > I feel that unsupervised learning is like this: I have run many classic unsupervised or self-supervised learning algorithms, but very few have achieved the results shown in the papers, and some even performed worse.
> > > > > > 
> > > > > > 
> > > > > > When reproducing the paper, it was found that the PSNR and SSIM of the paper first increased and then decreased, rather than converging towards the direction of increasing PSNR. Is this a common situation in self-supervised methods?
> > > > > 
> > > > > 
> > > > > Hello, I want to ask what PSNR value your training can reach. Why does the PSNR keep decreasing during my training, and it only ends up around 10 in the end?
> > > > 
> > > > 
> > > > We selected the checkpoint/epoch with the highest PSNR, as done in the paper. In my experiments, I reached performance close to the paper’s results around epoch 29, but the PSNR then gradually decreased. By the final epoch, it was about 17.
> > > 
> > > 
> > > May I ask if you modified the author's code? Why is it that when I train with LOLv1 in the same way, the PSNR starts at 13 and then just stays at 10? In my command line for training, I used , while the author's original command is . I removed all the other parameters from it—could that be the reason?`` python main.py --save_folder weights_3 --logroot logs/your_log/``python main.py --save_folder weights/LOLv1/ --logroot logs/your_log/ --data_train your_train_data --data_val your_val_data --referance_val the_high_quality_val_data --lr 1e-5 --light_patch 64 --loss_weights [1, 0.1, 0.1, 0.5] ``
> > 
> > 
> > 英文翻译如下：
> > > There might be some issues with your reproduction. When I reproduced the original code, the best result I could get was around **19 dB**, but it still didn’t reach the results reported in the paper. You need to modify the parameters of the **exposure loss**; follow the command provided by the authors. I’d recommend taking a look at  [#11](https://github.com/huaqlili/unsupervised-light-enhance-ICLR2025/issues/11).
> 
> But if I follow the author's instructions, options like --light_patch 64 and --loss_weights [1, 0.1, 0.1, 0.5] are not defined in the code, so running them would cause errors. Therefore, I multiplied the --loss_weights [1, 0.1, 0.1, 0.5] command-line parameter with the four losses in order, and I simply removed light_patch.
> 
> Also, may I ask what it means to change the exposure loss?
It is recommended to carefully review the code; this parameter should be the number of blocks in the exp loss, for supervising local brightness
